### PR TITLE
Added Missing use_federated_token Parameter to Azure Storage Configuration Docs for Workload Identity

### PR DESCRIPTION
**Pull request by SanderBlom**
**What this PR does**:
Added missing requirement for the usage of Azure Workload Identity. This parameter is specified [here](https://grafana.com/docs/tempo/latest/configuration/) under `Storage block configuration example`, but is not currently mentioned anywhere else.

Not providing this would result in a 403 response from the Azure Storage container when Tempo tries to access it.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by mapno — APPROVED** [review]
LGTM, thanks! I'll let @knylander-grafana do the final review.

File: docs/sources/tempo/configuration/azure.md [hunk 1]
```diff
@@ -10,15 +10,16 @@ weight: 30
 Tempo requires the following configuration to authenticate to and access Azure blob storage:
 
 - Storage Account name specified in the configuration file as `storage_account_name` or in the environment variable `AZURE_STORAGE_ACCOUNT`
-- Credentials for accessing the Storage Account; can be one of the following
+- Credentials for accessing the Storage Account; can be one of the following:
```
Comment (knylander-grafana) [inline docs/sources/tempo/configuration/azure.md:13]: Changing ; to a period to match the other bullets at the same level. 
```suggestion
- Credentials for accessing the Storage Account.  Can be one of the following:
```

File: docs/sources/tempo/configuration/azure.md [hunk 2]
```diff
@@ -10,15 +10,16 @@ weight: 30
...
   - Storage Account access key specified in the configuration file as `storage_account_key` or in the environment variable `AZURE_STORAGE_KEY`
   - An Azure Managed Identity; either system or user assigned. To use Azure Managed Identities, you'll need to set `use_managed_identity` to `true` in the configuration file or set `user_assigned_id` to the client ID for the managed identity you'd like to use.
       - For a system-assigned managed identity, no additional configuration is required.
       - For a user-assigned managed identity, you'll need to set `user_assigned_id` to the client ID for the managed identity in the configuration file.
```
Comment (knylander-grafana) [inline docs/sources/tempo/configuration/azure.md:17]: Updating to have it read a little better. 
```suggestion
- Credentials for accessing the Storage Account that are one of the following:
  - Storage Account access key specified in the configuration file as `storage_account_key` or in the environment variable `AZURE_STORAGE_KEY`
  - An Azure Managed Identity that is either system or user assigned. To use Azure Managed Identities, you'll need to set `use_managed_identity` to `true` in the configuration file or set `user_assigned_id` to the client ID for the managed identity you'd like to use.
      - For a system-assigned managed identity, no additional configuration is required.
      - For a user-assigned managed identity, you'll need to set `user_assigned_id` to the client ID for the managed identity in the configuration file.
```

**Review by joe-elliott — APPROVED** [review]
Nice. Thanks for the addition!

I'll let @knylander-grafana merge if she's good with the change, but lgtm.

**Comment by SanderBlom** [comment]
@knylander-grafana any other changes needed before merging?

**Comment by knylander-grafana** [comment]
I'm trying to resolve the conflicts in the PR. There are changes to headings in the same page that point to monolithic mode. The new content examples appear under these sections. Is this correct?

**Comment by knylander-grafana** [comment]
> @knylander-grafana any other changes needed before merging?

Sorry this took so long. I was out on PTO when this PR was filed. 

Pleas check the headings. I resolved a conflict and that means that your content is under slightly different heading.

File: docs/sources/tempo/configuration/azure.md [hunk 3]
```diff
@@ -10,15 +10,16 @@ weight: 30
...
+  - An Azure Managed Identity that is either system or user assigned. To use Azure Managed Identities, you'll need to set `use_managed_identity` to `true` in the configuration file or set `user_assigned_id` to the client ID for the managed identity you'd like to use.
       - For a system-assigned managed identity, no additional configuration is required.
       - For a user-assigned managed identity, you'll need to set `user_assigned_id` to the client ID for the managed identity in the configuration file.
-  - Via Azure Workload Identity. To use Azure Workload Identity, you'll need to enable Azure Workload Identity on your cluster, add the required label and annotation to the service account and the required pod label.
+  - Via Azure Workload Identity. To use Azure Workload Identity, you'll need to enable Azure Workload Identity on your cluster, add the required label and annotation to the service account and the required pod label. Additionally, you will need to set `use_federated_token` to `true` to utilize Azure Workload Identity.
 
 ## Sample configuration (for Tempo Monolithic Mode)
```
Comment (knylander-grafana) [inline docs/sources/tempo/configuration/azure.md:20]: This is one of the headings that was modified by the PR conflict resolution. Is your example okay in this section?

Comment (SanderBlom) [reply docs/sources/tempo/configuration/azure.md:20]: LGTM :smile:

**Comment by knylander-grafana** [comment]
Thank you all for the additional review verification. I'll merge this.

**Pull request merged**